src/rt/tracegc.d

```diff
@@ -91,65 +91,47 @@ extern (C) void[] _d_newarraymiTX(const TypeInfo ti, size_t[] dims);
 extern (C) void* _d_newitemT(in TypeInfo _ti);
 extern (C) void* _d_newitemiT(in TypeInfo _ti);
 
-extern (C) Object _d_newclassTrace(string file, int line, string funcname, const ClassInfo ci)
+private string generatePrintf ( )
 {
     version (tracegc)
     {
-        printf("_d_newclassTrace class = %s file = '%.*s' line = %d function = '%.*s'\n",
-            cast(char *)ci.name,
-            file.length, file.ptr,
-            line,
-            funcname.length, funcname.ptr
-            );
+        return q{
+            printf("%s file = '%.*s' line = %d function = '%.*s'\n",
+                __FUNCTION__.ptr,
+                file.length, file.ptr,
+                line,
+                funcname.length, funcname.ptr
+                );
+        };
```

Comment:
Why a `mixin` and not just a function with a `version` statement inside? The optimizer should remove the empty function call I guess...